### PR TITLE
Default to aot_eager for torch.compile on MPS

### LINKED ISSUE #96976: Compile gives weird error about CUDA on MPS device (apple Macbook)
### 🐛 Describe the bug

In PyTorch 2.0 on an Apple Macbook running latest OS.
```python
from torch import nn
from torchvision.models import resnet50

torch.set_default_device('mps')

img = torch.rand(1,3,64,64)
r50 = resnet50()

ccompiled_r50 = torch.compile(r50)
ccompiled_r50(img)
```
You get a weird error about `CUDA` ! 🤯
```
BackendCompilerFailed: debug_wrapper raised AssertionError: Torch not compiled with CUDA enabled

Set torch._dynamo.config.verbose=True for more information


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True
```

### Versions

Collecting environment information...
PyTorch version: 2.0.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.2.1 (arm64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.102)
CMake version: version 3.23.1
Libc version: N/A

Python version: 3.10.9 | packaged by conda-forge | (main, Feb  2 2023, 20:26:08) [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-13.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] numpy==1.24.2
[pip3] torch==2.0.0
[pip3] torchaudio==2.0.0
[pip3] torchvision==0.15.0
[conda] numpy                     1.24.2          py310h3d2048e_0    conda-forge
[conda] pytorch                   2.0.0                  py3.10_0    pytorch
[conda] torchaudio                2.0.0                 py310_cpu    pytorch
[conda] torchvision               0.15.0                py310_cpu    pytorch

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

### PR DESCRIPTION
**Pull request by msaroufim**
Fixes https://github.com/pytorch/pytorch/issues/96976

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/scheduler.py
Comment (albanD) [inline torch/_inductor/scheduler.py:1149]: Should we recommend `aot_eager` here?
Or should we swap the default on MPS to be aot_eager instead of inductor?

**Comment by msaroufim** [comment]
Hi @kulinseth also wanted to bring this issue to your attention, not sure if you've been working on a PT 2.0 integration https://github.com/apple/coremltools/issues/1752#issuecomment-1445439957

File: torch/_inductor/scheduler.py
Comment (albanD) [reply torch/_inductor/scheduler.py:1149]: Will let @kulinseth to make the final decision on this one.

File: torch/_inductor/utils.py
Comment (kulinseth) [inline torch/_inductor/utils.py:54]: Same change which Alban proposed above:
`torch.randn(1, device="mps")`

**Review by kulinseth — CHANGES_REQUESTED** [review]
Please add test cases and make the change to "aot_eager"

File: torch/_inductor/utils.py
Comment (albanD) [reply torch/_inductor/utils.py:54]: Ho we shouldn't need that call at all haha. Just use torch.backends.mps.is_available() !

File: torch/__init__.py
Comment (albanD) [inline torch/__init__.py:1448]: lint might not be happy about that :D

File: torch/testing/_internal/inductor_utils.py
Comment (albanD) [inline torch/testing/_internal/inductor_utils.py:23]: I think this will make internal linter unhappy and you'll get reverted ;) cc @ZainRizvi what's the status on making that lint consistent?

**Review by albanD — COMMENTED** [review]
Small nits but otherwise SGTM.

File: torch/testing/_internal/inductor_utils.py
Comment (msaroufim) [reply torch/testing/_internal/inductor_utils.py:23]: Linter has been acting weird today btw, these weird mypy errors showing up and it's also quite slow - not sure if thats' related

Comment (ZainRizvi) [reply torch/testing/_internal/inductor_utils.py]: Lint consistency work was merged in a couple weeks ago.

To take effect, it requires updating linter on the local dev machine by running `pip install lintrunner -U` (and you should be getting a warning/nudge to do so whenever you run `lintrunner` locally)

**Comment by msaroufim** [comment]
@kulinseth lmk if this looks good to merge

**Review by kulinseth — APPROVED** [review]
Looks good.

**Comment by msaroufim** [comment]
@albanD could I get 1 more stamp plz

File: torch/__init__.py
Comment (msaroufim) [reply torch/__init__.py]: Just checked after `pip install lintrunner -a` and no lint issues

**Review by albanD — APPROVED** [review]
Sounds ok. The default value problem can be resolved later I guess?

File: torch/__init__.py
Comment (msaroufim) [reply torch/__init__.py:1445]: Yeah I think we chose early on to avoid having to tell users to `torch.compile(backend="inductor")` - the whole API needs more work, I can take that on next

**Review by ZainRizvi — APPROVED** [review]

**Comment by albanD** [comment]
btw @msaroufim the CI failures definitely could be related. You want to double check that before merging.

**Comment by msaroufim** [comment]
@pytorch merge -r

**Comment by khaled-alkilane** [comment]
Has it been solved?? How can we use 'torch.compile' now. Do one need to upgrade Pytorch?

**Comment by msaroufim** [comment]
Yeah you're going to need to upgrade to latest nightly tomorrow. In the meantime use backend=aot_eager as a workaround if you can't upgrade yet

**Pull request closed without merging**
